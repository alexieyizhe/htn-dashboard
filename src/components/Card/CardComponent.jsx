import React from "react";
import styled from "styled-components";

/* Images */
import CheckmarkImg from "../../../static/checkmark.png";


const CardContainer = styled.div`
  width: 60vw;
  height: 10vw;
  padding: 2vw;

  display: flex;
  flex-direction: row;
  justify-content: space-between;

  color: white;
  background-color: ${props => props.completed ? props.theme.colors.green : props.theme.colors.blue};

  border-radius: ${props => props.theme.defaults.borderRadius};
  cursor: pointer;

  will-change: transform;
  transition: transform 250ms ease-in-out;
  transform: scale(1.0);
  &:hover {
    transform: scale(1.02);
  }
`;


const CardLabel = styled.div`
  font-size: 1.5em;
`;


const CardCheckmark = styled.img`
  max-height: 50%;
  margin: auto 1vw; // centers img
`;


const Card = ({
    className,
    label = "DEFAULT LABEL",
    completed = false,
    backgroundImg,
    clickHandler = () => console.log(`You clicked ${label}`),
}) => (
  <CardContainer
    className={className}
    completed={completed}
    backgroundImg={backgroundImg}
    onClick={() => clickHandler()}
  >
    <CardLabel>{label}</CardLabel>
    {completed && <CardCheckmark src={CheckmarkImg} alt={`Checkmark indicating completion of the ${label} question set.`} />}
  </CardContainer>
);


export default Card;
