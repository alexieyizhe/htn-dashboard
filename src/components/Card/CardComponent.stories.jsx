import React from "react";
import { storiesOf } from "@storybook/react";
import Card from "./CardComponent";


storiesOf("Card", module)
  .add("Default", () => <Card />)
  .add("Completed", () => <Card completed />);
