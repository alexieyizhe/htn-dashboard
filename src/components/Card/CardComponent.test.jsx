/* eslint-disable no-undef */
import React from "react";

import Card from "./CardComponent";

describe("Card", () => {
  it("renders correctly", () => {
    const cardComponent = mountWithTheme(<Card siteTitle="Default Starter" />);
    expect(cardComponent).toMatchSnapshot();
  })
})
