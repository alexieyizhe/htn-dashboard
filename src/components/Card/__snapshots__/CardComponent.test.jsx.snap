// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
.c0 {
  width: 60vw;
  height: 10vw;
  padding: 2vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  color: white;
  background-color: #68B0E5;
  border-radius: 10px;
  cursor: pointer;
  will-change: transform;
  -webkit-transition: -webkit-transform 250ms ease-in-out;
  -webkit-transition: transform 250ms ease-in-out;
  transition: transform 250ms ease-in-out;
  -webkit-transform: scale(1.0);
  -ms-transform: scale(1.0);
  transform: scale(1.0);
}

.c0:hover {
  -webkit-transform: scale(1.02);
  -ms-transform: scale(1.02);
  transform: scale(1.02);
}

.c1 {
  font-size: 1.5em;
}

<Card
  siteTitle="Default Starter"
>
  <styled.div
    completed={false}
    onClick={[Function]}
  >
    <StyledComponent
      completed={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  width: 60vw;
  height: 10vw;
  padding: 2vw;

  display: flex;
  flex-direction: row;
  justify-content: space-between;

  color: white;
  background-color: ",
              [Function],
              ";

  border-radius: ",
              [Function],
              ";
  cursor: pointer;

  will-change: transform;
  transition: transform 250ms ease-in-out;
  transform: scale(1.0);
  &:hover {
    transform: scale(1.02);
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
    >
      <div
        className="c0"
        onClick={[Function]}
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  font-size: 1.5em;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            >
              DEFAULT LABEL
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</Card>
`;
